providers/ethereum-provider/src/EthereumProvider.ts

```diff
@@ -160,31 +161,40 @@ export function buildNamespaces(params: NamespacesParams): {
   const optional: Namespace = {
     chains: [
       ...new Set(
-        shouldIncludeRequiredChains ? requiredChains.concat(optionalChains || []) : optionalChains,
+        shouldIncludeRequiredChains ? required.chains.concat(optionalChains || []) : optionalChains,
+      ),
+    ],
+    methods: [
+      ...new Set(
+        required.methods.concat(optionalMethods?.length ? optionalMethods : OPTIONAL_METHODS),
```

Comment:
Not sure if intentional so flagging: seems like we're now always adding `OPTIONAL_METHODS` to methods if `optionalMethods` is not defined. 

Makes sense to me but is a deviation from the previous `[]`

Reply:
Question for me is: shouldn't we then do the same with `OPTIONAL_EVENTS`?